templates/core/terraform/network/dns_zones.tf

```diff
@@ -257,6 +247,150 @@ resource "azurerm_private_dns_zone_virtual_network_link" "eventgridlink" {
   lifecycle { ignore_changes = [tags] }
 }
 
+resource "azurerm_private_dns_zone" "purview_account" {
```

Comment:
Instead of copy/pasting all these new sections... have you considered using the for_each method already present at the bottom of this file?

Reply:
I hadn't! that is much nicer – thanks